docs/compilers/Co-located core types.md

```diff
@@ -15,7 +15,7 @@ Scenarios
 
 ## Design
 
-Compilers may assume that there is exactly one assembly representing the "core assembly". This core assembly is either referenced or is currently being compiled.
+Compilers may assume that there is exactly one assembly representing the "core assembly". This core assembly is either referenced or is currently being compiled. The unique core assembly may be identified by the compiler as the first referenced assembly that does not reference any other assembly, or the assembly being compiled if there are no referenced assemblies.
 
```

Comment:
The precise rules that we use is that we look for an assembly that satisfies the following constraints, and require there be a unique candidate ([CommonReferenceManager.Binding.cs line 897](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Binding.cs#L897)):
- May not be a linked reference
- Must not reference any other assemblies
- Must not contain any NoPia types
- Some other condition about not being superseded
- Defines System.Object

The native compiler allowed more than one candidate and selected the first one, but we never reproduced that behavior in Roslyn.